rpc-broker/src/rpc-json.c

```diff
@@ -0,0 +1,348 @@
+    //      (possibly adding union-type in the `struct arg`)
+
+    switch (type) {
+
+        case ('b'): {
+            int json_bool = json_object_get_boolean(jarg);
+            args[idx] = malloc(sizeof(int));
+            memcpy(args[idx], (void *) &json_bool, sizeof(int));
+            break;
+        }
+
+        case ('u'):
+        case ('i'): {
+            int json_int = json_object_get_int(jarg);
+            args[idx] = malloc(sizeof(int));
+            memcpy(args[idx], (void *) &json_int, sizeof(int));
+            break;
+        }
+
+        case ('d'): {
+            double json_double = json_object_get_double(jarg);
+            args[idx] = malloc(sizeof(double));
+            memcpy(args[idx], (void *) &json_double, sizeof(double));
+            break;
+        }
+
+        case ('s'): {
+            const char *json_str = json_object_get_string(jarg);
+            if (!json_str)
+                json_str = "";
+            args[idx] = strdup(json_str);
+            break;
+        }
+
+        case ('v'): {
+            int jtype = json_object_get_type(jarg);
+            type = json_dbus_types[jtype];
+            append_dbus_message_arg(type, idx, args, jarg);
+            break;
+        }
+
+        default:
+            break;
+    }
+}
+
+static const char *get_json_str_obj(struct json_object *jobj, char *field)
+{
+    struct json_object *jfield;
+
+    if (!json_object_object_get_ex(jobj, field, &jfield))
+        return NULL;
+
+    return strdup(json_object_get_string(jfield));
+}
+
+void load_json_response(DBusMessage *msg, struct json_response *jrsp)
+{
+    DBusMessageIter iter, sub;
+    char *sig;
+    struct json_object *args;
+
+    dbus_message_iter_init(msg, &iter);
+
+    sig = dbus_message_iter_get_signature(&iter);
+    snprintf(jrsp->arg_sig, DBUS_MAX_ARG_LEN - 1, "%s", sig);
+    dbus_free(sig);
+
+    args = jrsp->args;
+
+    if (jrsp->arg_sig && jrsp->arg_sig[0] == 'a') {
+
+        dbus_message_iter_recurse(&iter, &sub);
+        iter = sub;
+        // This accomodates the legacy code in `rpc-proxy` where the
+        // signatures are being mis-handled and thus the UI is excpecting
+        // the malformed dbus responses.
+        switch (jrsp->arg_sig[1]) {
+            case 'a':
+            case 'i':
+            case 's':
+            case 'o': {
+                struct json_object *array = json_object_new_array();
+                json_object_array_add(jrsp->args, array);
+                args = array;
+                break;
+            }
+
+            default:
+                break;
+        }
+    }
+
+    parse_signature(args, NULL, &iter);
+}
+
+static signed int parse_json_args(struct json_object *jarray,
+                                  struct json_request *jreq)
+{
+    char *signature, *sigptr;
+    size_t array_length;
+    int i, jtype;
+    struct json_object *jarg;
+
+    // XXX supports the removal of network-daemon/slave
```

Comment:
"XXX" looks bad. Is "supportS" a typo, did you mean "TODO: Support the removal [...]"?

Reply:
No typo, I should have been clearer.  With that condition, it shows its a message thats there because NWD/NWS are removed.